client/src/app/appCtrl.js

```diff
@@ -46,36 +60,127 @@ angular.module('app').controller('appCtrl', ['$scope', '$http', '$timeout', 'app
     slidingWindow = false;
     $scope.$broadcast("fileUploadChange");
     $scope.view.loading = true;
-    // we do a quick test here to see if the server supports the Range header.
-    // If so, we try to stream. If not, we try to download.
-    $http.head($scope.view.filePath,{'headers' : {'Range' : 'bytes=0-32'}}).then(function(response){
-      if(response.status === 206) {
-        // now we check to see how big the file is
-        $http.head($scope.view.filePath).then(function(response){
-          var contentLength = response.headers('Content-Length');
-          if (contentLength > appConfig.MAX_FILE_SIZE) {
-            slidingWindow = true;
-            $scope.view.windowing.show = true;
-          }
-          streamRemoteFile($scope.view.filePath);
-        });
-      } else {
-        downloadFile($scope.view.filePath);
-      }
-    }, function() {
-      downloadFile($scope.view.filePath);
-    });
+    $scope.view.file.size = getFileSize($scope.view.filePath);
```

Comment:
thanks for explanation! although I'm not very clever from it :wink: I'll try to study that, or if you can come up with a `getFileSize` for HTTP headers?